test/puppetlabs/dujour/version_check_test.clj

```diff
@@ -122,6 +122,34 @@
           (is (nil? (message "certname")))
           (is (nil? (message "cacert")))))))
 
+  (testing "sends agent_os instead of agent-os"
+    (with-test-logging
+      (jetty9/with-test-webserver return-all-as-message-app port
+        (let [return-val (promise)
+              callback-fn (fn [resp] (deliver return-val resp))
+              agent-os {"debian" 15 "centos" 5}
+              _ (check-for-updates! {:agent-os agent-os
+                                     :product-name     "foo"
+                                     :database-version "9.4"}
+                                    (format "http://localhost:%s" port) callback-fn)
+              message (json/parse-string (:message @return-val))]
+          (is (= (message "agent_os") agent-os))
+          (is (nil? (message "agent-os")))))))
+
+  (testing "doesn't clobber agent_os"
+    (with-test-logging
+      (jetty9/with-test-webserver return-all-as-message-app port
+        (let [return-val (promise)
+              callback-fn (fn [resp] (deliver return-val resp))
+              agent_os {"debian" 15 "centos" 5}
+              _ (check-for-updates! {:agent_os agent_os
```

Comment:
The only other case I think we might want to watch for (and maybe this is overkill since we're the consumers of this library) is if someone supplies both agent-os and agent_os.